### ISSUE TITLE
PLIP: new changelog format

### ISSUE THREAD

**Issue by gforcada** [plone/Products.CMFPlone#1180]
Current changelog (on ``CHANGES.rst``) are just a plain list of items. When a new Plone release is shapping up @esteele sends a huge combined changelog, that's really nice but not so practical for non-core developers, specially for docs and marketing team for instance.

The idea would be to standardize a new format that would help spot changes that probably matter for marketing, documentation and other teams.

See Firefox release notes for instance: [42.0beta](https://www.mozilla.org/en-US/firefox/42.0beta/releasenotes), [41](https://www.mozilla.org/en-US/firefox/41.0/releasenotes) they split the main changes into ``New features``, ``Changed``, ``HTML5``, ``Bugs fixed``, ``Developers``.

Or Couch potato: https://couchpota.to/updates/changelog.html

An initial proposal could be:

```
- New features

  - *add item here*

- Bugs fixed

  - *add item here*

- Docs

  - *add item here*

- Cleanup

  - *add item here*
```

So basically the format stays the same just that there are 4 top-level categories where to add you change log entry.

Not blocked by, but it would be great to "merge" together with the ``CONTRIBUTORS.rst`` https://github.com/plone/Products.CMFPlone/issues/1179

Note that [zest.releaser](https://github.com/zestsoftware/zest.releaser) would need to be updated (or plugins for it written)

Any seconders?

**Comment by hvelarde** [comment]
+½ as stated in #1179, I see no advantage on adding Docs and Cleanup sections.

**Comment by gforcada** [comment]
Cleanup's purpose is quite simple: every change related to code cleanups, internal refactorings and so on should go there. If p.a.dexterity reorganizes its code internally but keeps providing the same API is not a feature nor a bug nor a documentation change, but is still important to add a note about it because as python allows to import and use any method of any class, someone will still be importing and using some internal methods and *boom*, all of a sudden it's broken. Sure one should not use those to begin with, but a note on the changelog will help you know where to look at.

As for the docs: please tell the docs team that documentation changes are not worth being added on the changelog, if they agree with that I'm fine. On the other hand, for some packages which do not have documentation is ok to leave it empty. The release team will already clean that up prior to making a release.

**Comment by mauritsvanrees** [comment]
+1

We can make it a bit smaller, to avoid developers losing sleep over where exactly an entry needs to be:

```
- New:

  - *add item here*

- Fixes:

  - *add item here*
```

An example can then be:

```
- New:

  - Added documentation on Frob5000 feature introduced in previous release.
    [maurits]

- Fixes:

  - Fixed pep8 warnings.
    [gforcada]
```

To add this in the postrelease phase of zest.releaser his needs an entry point in plone.releaser to set the exact text we want. See `nothing_changed_yet` in http://zestreleaser.readthedocs.org/en/latest/entrypoints.html. Looks like this may need a few minor changes in zest.releaser, as it currently expects a single line. We probably want to be able to set the text that is added in postrelease and a separate text that is checked in prerelease: warn when `*add item here*` is still in the text.
I can assist.

**Comment by gforcada** [comment]
@mauritsvanrees my idea of separate new/features/docs is mostly to help marketing/non-technicals to see the most prominent changes.

They would probably just skimm over the huge combined change log and look for the "new features" section.

Or to put it the other way around: things on "new features" (other name suggestions are welcome) would be potential candidates to get on the release notes, be highlighted on the social media / newsletter, etc etc

**Comment by mauritsvanrees** [comment]
True. Of course every bug fixed can be interesting too ("hey, did you encounter this bug when doing X? It now works.") Same for documentation added ("ever wondered how X worked, find out here"). But new features are the most interesting fodder for shouting about in communications.

**Comment by jensens** [comment]
+1 for @mauritsvanrees solution with two simple section ``new`` and ``fixes``. Cleanup goes under fixes and docs can be a fix or something new too, or new code with docs. That works, is simple enough and make reading a bit easier for non-coredevs.

**Comment by mauritsvanrees** [comment]
I have updated plone.releaser (which you get by doing `bin/buildout -c release.cfg` in coredev 5) to write the new changelog format.

I have released an example package with this: https://pypi.python.org/pypi/mauritstestpackage
Note that this is what you get when you do not actually edit the changelog file yourself and do a release anyway.
This is the postrelease commit: https://github.com/mauritsvanrees/mauritstestpackage/commit/e93736729a6be74c4122eb6b6e8b057da22084d1

Note that when I now do a prerelease on the test package, I get a different warning:

```
$ ~/community/plone-coredev/5.0/bin/prerelease
Run pyroma on the package before tagging? (Y/n)? n
Do you want to run check-manifest? (Y/n)? n
Enter version [1.6]: 
WARNING: Changelog contains '*add item here*'. Are you sure you want to release? (y/N)? 
INFO: You can use the 'lasttaglog' command to see the commits since the last tag.
```

Also note that we will now *not* get warned when a package has the old format and still has 'Nothing changed yet' in its changelog.

**Comment by hvelarde** [comment]
plone.releaser? is that a new package?

**Comment by mauritsvanrees** [comment]
No, we (the release team) have been using this for a while now for the core development packages.  It does a few extra checks and it updates the checkouts.cfg and versions.cfg in coredev after releasing a package.
See https://github.com/plone/plone.releaser

**Comment by gforcada** [comment]
@mauritsvanrees cool, thanks a lot!

**Comment by mauritsvanrees** [comment]
I have added a pull request to zest.releaser that gives us a hook to specify some texts in plone.releaser that we require in the prerelease phase.  We can make it so that at least one of 'New:' or 'Fixes:' has to be there.  Well, when neither is there, zest.releaser will ask us if we want to continue, with default answer 'No', so you can still ignore the advice if you want to.
See https://github.com/zestsoftware/zest.releaser/pull/139

**Comment by mauritsvanrees** [comment]
I have updated `plone.releaser` with an extra check, which now requires `zest.releaser` 6.0.

Sample run:

```
$ ~/community/plone-coredev/5.0/bin/prerelease 
Run pyroma on the package before tagging? (Y/n)? n
Do you want to run check-manifest? (Y/n)? n
Enter version [3.0.13]: 
WARNING: Changelog should contain at least one of these required strings: "New:", "Fixes:". Are you sure you want to release? (y/N)? 
```

**Comment by mauritsvanrees** [comment]
I did a bulk update of all packages on coredev 5, adding New/Fixes headings.
I skipped packages that did not have a `CHANGES.rst`.
